packages-ui/roosterjs-react/lib/emoji/components/EmojiIcon.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { Emoji } from '../type/Emoji';
+import { memoizeFunction } from '@fluentui/react/lib/Utilities';
+import { mergeStyleSets } from '@fluentui/react/lib/Styling';
+import { Theme, useTheme } from '@fluentui/react/lib/Theme';
+
+/**
+ * @internal
+ * Emoji icon data
+ */
+export interface EmojiIconProps {
+    id: string;
+    emoji: Emoji;
+    strings: Record<string, string>;
+    onClick?: (e: React.MouseEvent<EventTarget>) => void;
+    onMouseOver?: (e: React.MouseEvent<EventTarget>) => void;
+    onFocus?: React.FocusEventHandler<HTMLButtonElement>;
+    isSelected?: boolean;
+}
+
+/**
+ * @internal
+ * Emoji icon component
+ */
+export default function EmojiIcon(props: EmojiIconProps) {
+    const { emoji, onClick, isSelected, onMouseOver, onFocus, strings, id } = props;
+    const content = strings[emoji.description];
+    const theme = useTheme();
+    const classNames = getEmojiIconClassNames(theme);
+    return (
+        <button
+            id={id}
+            role="option"
+            className={classNames.emoji}
+            onClick={onClick}
+            onMouseOver={onMouseOver}
+            onFocus={onFocus}
+            data-is-focusable={true}
+            aria-label={content}
+            aria-selected={isSelected}
+            {...getDataAndAriaProps(props)}>
+            {emoji.codePoint || '…'}
+        </button>
+    );
+}
+
+const getEmojiIconClassNames = memoizeFunction((theme: Theme) => {
```

Comment:
This function will be called every time when EmojiIcon is rendered, which means it will be called hundreds of times. Suggest move it out of this file into EmojiPake.tsx, and just pass in the class name, so it will only be called once.

Reply:
Same for other files. Let's only have one mergeStyleSet() for emoji, and just pass in class names to other inner components